notebook_connector/secret_store.py

```diff
@@ -0,0 +1,161 @@
+import contextlib
+import logging
+import os
+from dataclasses import dataclass
+from sqlcipher3 import dbapi2 as sqlcipher
+from typing import List, Optional, Union
+from inspect import cleandoc
+
+
+_logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class Table:
+    name: str
+    columns: List[str]
+
+
+SECRETS_TABLE = Table("secrets", ["user", "password"])
+CONFIG_ITEMS_TABLE = Table("config_items", ["item"])
+
+
+@dataclass(frozen=True)
+class Credentials:
+    user: str
+    password: str
+
+
+class InvalidPassword(Exception):
+    """Signal potentially incorrect master password."""
+
+
+class Secrets:
+    def __init__(self, db_file: str, master_password: str) -> None:
+        self.db_file = db_file
+        self._master_password = master_password
+        self._con = None
+
+    def close(self) -> None:
+        if self._con is not None:
+            self._con.close()
+            self._con = None
+
+    def connection(self) -> sqlcipher.Connection:
+        if self._con is None:
+            db_file_found = os.path.exists(self.db_file)
```

Comment:
I prefer pathlib.Path.exists, the os package is not recommended anymore (can be done in a later PR)